python/src/server.py

```diff
@@ -0,0 +1,65 @@
+from konan_sdk.konan_service.services import KonanService
+from konan_sdk.konan_service.models import KonanServiceBaseModel
+from konan_sdk.konan_service.serializers import (
+    KonanServiceBasePredictionRequest, KonanServiceBasePredictionResponse)
+
+
+class MyPredictionRequest(KonanServiceBasePredictionRequest):
+    """Defines the schema of a prediction request
+
+    Follow the convention of <field_name>: <type_hint>
+    Check https://pydantic-docs.helpmanual.io/usage/models/ for more info
+
+    Optionally add validators for your features
+    Follow the pydantic convention
+    Check https://pydantic-docs.helpmanual.io/usage/validators/ for more info
+    """
+    # TODO: [1] Define your request schema
+    pass
+
+
+class MyPredictionResponse(KonanServiceBasePredictionResponse):
+    """Defines the schema of a prediction response
+
+    Follow the convention of <field_name>: <type_hint>
+    Check https://pydantic-docs.helpmanual.io/usage/models/ for more info
+
+    Optionally add validators for your features
+    Follow the pydantic convention
+    Check https://pydantic-docs.helpmanual.io/usage/validators/ for more info
+    """
+    # TODO: [2] Define your response schema
+    pass
+
+
+class MyModel(KonanServiceBaseModel):
+    def __init__(self):
+        """Add logic to initialize your actual model here
+
+        Maybe load weights, connect to a database, etc ..
+        For example, the following code will load a model saved as a model.pickle file in the models/ directory
+        
+        import pickle
+        from konan_sdk.konan_service import constants as Konan_Service_Constants
+        self.loaded_model = pickle.load(open(f"{Konan_Service_Constants.MODELS_DIR}/model.pickle", 'rb'))
+        """
+        super().__init__()
+        # TODO: [3] Initialize your model
+
+    def predict(self, req: MyPredictionRequest) -> MyPredictionResponse:
```

Comment:
Will the user need to call the response or request serializer? Or they just return dictionaries and fastapi will do the serialization.

Reply:
`fastapi` will pass a `MyPredictionRequest` object to the endpoint, and we forward it to the `MyModel.predict()` method.
Users should have the `MyModel.predict()` method return a `MyPredictionResponse` object, which our endpoint will then also return. `fastapi` will handle the serialization automatically to the corresponding JSON.

Reply:
Where is the point you instruct the user to return `MyPredictionResponse` object from models predict endpoint?

Reply:
In addition to the plain-text explanation in `konan-docs`, this is also defined in [here](https://github.com/SynapseAnalytics/konan-template-deployments/blob/KONAN-764/python/src/server.py#L49) and [here](https://github.com/SynapseAnalytics/konan-template-deployments/blob/KONAN-764/python/src/server.py#L56)

Reply:
I get they are defined there,  but I don't think this is enough,  As a user I don't get from the type hint that my response,  from predict endpoint should be of this signature.
I get it may seem obvious, but we need to make it clear,  maybe we discuss this in another issuem?